lib/net.js

```diff
@@ -827,7 +827,8 @@ function connect(self, address, port, addressType, localAddress, localPort) {
     var sockname = self._getsockname();
     var details;
 
-    if (sockname) {
+    if (sockname && sockname.address !== undefined
```

Comment:
I agree. Just not next to my laptop or I'd just type out the code exactly how it should appear stylistically. e.g. && always at the end, new line for each conditional.